### PR TITLE
add folding command

### PR DESCRIPTION
**Pull request by rebornix**
Support following commands

* `z o`
* `z c`
* `z M`
* `z R`

### CODE REVIEW

File: src/Actions/Fold.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {commands} from 'vscode';
+import {ActionSelection} from './Selection';
```
Comment (aioutecism) [inline src/Actions/Fold.ts]: This is not used and will be marked as error after we update TypeScript to 2.0.3. See #138 .

**Comment by aioutecism** [comment]
Cool!
Thanks for your contribution!

File: src/Actions/Fold.ts (hunk 1 above)
Comment (rebornix) [inline src/Actions/Fold.ts]: ha, that's copied directly from another file and should be moved.

**Comment by rebornix** [comment]
@aioutecism comments addressed.

**Comment by aioutecism** [comment]
I noticed that Vim accepts folding commands in Visual and VisualLine modes.
Cloud you add that too?

**Comment by rebornix** [comment]
@aioutecism please take a look :)

**Comment by aioutecism** [comment]
Thanks!

**Pull request merged**